source/common/upstream/health_discovery_service.cc

```diff
@@ -189,6 +192,18 @@ void HdsDelegate::onReceiveMessage(
   stats_.requests_.inc();
   ENVOY_LOG(debug, "New health check response message {} ", message->DebugString());
 
+  // Validate message fields
+  try {
+    MessageUtil::validate(*message, validation_visitor_);
+  } catch (const ProtoValidationException& ex) {
+    // Increment error count
+    stats_.errors_.inc();
+    ENVOY_LOG(warn, "unable to validate hds proto: {}", ex.what());
+
+    // Do not continue processing message
+    return;
+  }
+
```

Comment:
I wonder if we still want to keep ticking by issuing a `setHdsStreamResponseTimer()` as below with the latched `server_response_ms_`? This would ensure we keep reporting with previous valid config.

Reply:
Harvey, I went ahead and checked this one out. It looks like this is a good idea, as long as there is a previous valid specifier that was received. I added a check to see if there previously was valid clusters, and in this case enables the timer based on the previous value. I also added a unit test to have one successful call with a valid specifier followed by an invalid one, which falls back upon failing the validation check.

Reply:
Harvey,
Not sure if I missed anything. If hds_clsters_ was not empty, it was created by previous onReceiveMessage, which should already enabled the timer, unless the previous valid config has a interval=ZeroDuration.
The code below only calls setHdsStreamResponseTimer() when server_response_ms is different from the existing one.
For this case, do we want to use the new server_response_ms if different?
Thanks!

Reply:
Actually @xdzhai is right. This behavior should have been working before. No need to adjust `server_response_ms_` is we have an invalid config received.

Reply:
@htuch yes, it seems I looked over the timer enable call at the end of the sendResponse call. I've removed the redundant timer enable call as a result and use your suggestion about checking timer statuses instead.